app/controllers/work_packages/activities_tab_controller.rb

```diff
@@ -0,0 +1,247 @@
+      ),
+      layout: false
+    )
+  end
+
+  def update_filter
+    filter = params[:filter]&.to_sym || :all
+
+    update_via_turbo_stream(
+      component: WorkPackages::ActivitiesTab::Journals::FilterAndSortingComponent.new(
+        work_package: @work_package,
+        filter:
+      )
+    )
+    update_via_turbo_stream(
+      component: WorkPackages::ActivitiesTab::Journals::IndexComponent.new(
+        work_package: @work_package,
+        filter:
+      )
+    )
+
+    respond_with_turbo_streams
+  end
+
+  def update_streams
+    generate_time_based_update_streams(params[:last_update_timestamp], params[:filter])
+
+    respond_with_turbo_streams
+  end
+
+  def edit
+    # check if allowed to edit at all
+    update_via_turbo_stream(
+      component: WorkPackages::ActivitiesTab::Journals::ItemComponent.new(
+        journal: @journal,
+        state: :edit,
+        filter: params[:filter]&.to_sym || :all
+      )
+    )
+
+    respond_with_turbo_streams
+  end
+
+  def cancel_edit
+    update_via_turbo_stream(
+      component: WorkPackages::ActivitiesTab::Journals::ItemComponent.new(
+        journal: @journal,
+        state: :show,
+        filter: params[:filter]&.to_sym || :all
+      )
+    )
+
+    respond_with_turbo_streams
+  end
+
+  def create
+    call = create_notification_service_call
+
+    if call.success? && call.result
+      if call.result.initial?
+        # we need to update the whole item component for an initial journal entry
+        # and not just the show part as happens in the time based update
+        # as this part is not rendered for initial journal
+        update_item_component(call.result, state: :show, filter: params[:filter]&.to_sym || :all)
+      end
+      generate_time_based_update_streams(params[:last_update_timestamp], params[:filter])
+    end
+
+    respond_with_turbo_streams
+  end
+
+  def update
+    call = Journals::UpdateService.new(model: @journal, user: User.current).call(
+      notes: journal_params[:notes]
+    )
+
+    if call.success? && call.result
+      update_item_component(call.result, state: :show, filter: params[:filter]&.to_sym || :all)
+    end
+    # TODO: handle errors
+
+    respond_with_turbo_streams
+  end
+
+  def update_sorting
+    filter = params[:filter]&.to_sym || :all
+
+    call = Users::UpdateService.new(user: User.current, model: User.current).call(
+      pref: { comments_sorting: params[:sorting] }
+    )
+
+    if call.success?
+      # update the whole tab to reflect the new sorting in all components
+      # we need to call replace in order to properly re-init the index stimulus component
+      replace_via_turbo_stream(
+        component: WorkPackages::ActivitiesTab::IndexComponent.new(
+          work_package: @work_package,
+          filter:
+        )
+      )
+    end
+
+    respond_with_turbo_streams
+  end
+
+  private
+
+  def find_work_package
+    @work_package = WorkPackage.find(params[:work_package_id])
```

Comment:
All these `find...` methods need error handling for the turbo stream case. In a regular call this would look like:

```ruby
def find_work_package
  @work_package = ....
rescue ActiveRecord::RecordNotFound
  render_404
end
```

We need something similar here, maybe to return a toast?

Reply:
Done